File: Projet/Backend/src/main/java/com/example/filrouge_loubna/entities/MedicalOfficeStaff.java
package com.example.filrouge_loubna.entities;

import lombok.AllArgsConstructor;
import lombok.Data;
import lombok.NoArgsConstructor;
import lombok.ToString;

import javax.persistence.*;
import java.io.Serializable;

@Data
@Entity
@Table(name = "MedicalOfficeStaff")
@AllArgsConstructor
@NoArgsConstructor
@ToString
public class MedicalOfficeStaff implements Serializable {

    private static final long serialVersionUID = 1L;

    @Id
    @Column(name = "id", nullable = false)
    private Long id;


    @OneToOne(fetch=FetchType.LAZY,cascade = CascadeType.REMOVE)
    private User medicalOfficeStaffUser;

    private Long medicalOfficeId;
}

File: Projet/Frontend/src/app/models/MedicalOfficeStaff.ts
import { MedicalOffice } from "./MedicalOffice";
import { User } from "./User";

export interface MedicalOfficeStaff
{
    id?              : number,
    user?            : User,
    medicalOfficeStaffUser?            : User,
    medicalOfficeId? : number,
    medicalOffice?   : MedicalOffice
}